cloudinit/util.py

```diff
@@ -1818,15 +1844,10 @@ def uptime():
                 uptime_str = contents.split()[0]
         else:
             method = 'ctypes'
-            libc = ctypes.CDLL('/lib/libc.so.7')
-            size = ctypes.c_size_t()
-            buf = ctypes.c_int()
-            size.value = ctypes.sizeof(buf)
-            libc.sysctlbyname("kern.boottime", ctypes.byref(buf),
-                              ctypes.byref(size), None, 0)
+            # This is the *BSD codepath
             now = time.time()
-            bootup = buf.value
-            uptime_str = now - bootup
+            bootup = boottime()
```

Comment:
the uptime implementation obtains the boottime, and then the now timestamp.  As it is now, we capture now value before running boottime() so that's a slight change in implementation.  Let's restore the original order and move the boottime() call right before the time.time() call and we can drop the temp variables along the way.

```
# This is the *BSD codepath
uptime_str = str(time.time() - boottime())
```